internal/factsengine/gatherers/sbddump_test.go

```diff
@@ -0,0 +1,170 @@
+package gatherers_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	utilsMocks "github.com/trento-project/agent/pkg/utils/mocks"
+	"github.com/trento-project/agent/test/helpers"
+)
+
+type SBDDumpTestSuite struct {
+	suite.Suite
+}
+
+func TestSBDDumpTestSuite(t *testing.T) {
+	suite.Run(t, new(SBDDumpTestSuite))
+}
+
+func (suite *SBDDumpTestSuite) TestSBDDumpUnableToLoadDevices() {
+	mockExecutor := new(utilsMocks.CommandExecutor)
+	sbdDumpGatherer := gatherers.NewSBDDumpGatherer(
+		mockExecutor,
+		helpers.GetFixturePath("discovery/cluster/sbd/sbd_config_invalid"),
+	)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "sbd_devices_dump",
+			Gatherer: "sbd_dump",
+		},
+	}
+
+	gatheredFacts, err := sbdDumpGatherer.Gather(factRequests)
+
+	expectedError := &entities.FactGatheringError{
+		Message: "error loading the configured sbd devices: could not parse sbd config file: error on line 1: missing =",
+		Type:    "sbd-devices-loading-error",
+	}
+	suite.EqualError(err, expectedError.Error())
+	suite.Empty(gatheredFacts)
+}
+
+func (suite *SBDDumpTestSuite) TestSBDDumpUnableToDumpDevice() {
+	mockExecutor := new(utilsMocks.CommandExecutor)
+
+	mockOutputFile, _ := os.Open(helpers.GetFixturePath("gatherers/dev.vdc.sbddump.output"))
+	mockOutput, _ := ioutil.ReadAll(mockOutputFile)
+	mockExecutor.On("Exec", "sbd", "-d", "/dev/vdb", "dump").Return(nil, errors.New("a failure"))
+	mockExecutor.On("Exec", "sbd", "-d", "/dev/vdc", "dump").Return(mockOutput, nil)
+
+	sbdDumpGatherer := gatherers.NewSBDDumpGatherer(
+		mockExecutor,
+		helpers.GetFixturePath("discovery/cluster/sbd/sbd_config"),
+	)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "sbd_devices_dump",
+			Gatherer: "sbd_dump",
+		},
+		{
+			Name:     "another_sbd_devices_dump",
+			Gatherer: "sbd_dump",
+			Argument: "an-ignored-argument",
+		},
+	}
+
+	gatheredFacts, err := sbdDumpGatherer.Gather(factRequests)
+
+	expectedFacts := []entities.Fact{
+		{
+			Name:  "sbd_devices_dump",
+			Value: nil,
+			Error: &entities.FactGatheringError{
+				Message: "error while executing sbd dump: a failure",
```

Comment:
I think here the desired output would to show which device failed. So maybe we could update the wrapped or the returned error somehow